### PR TITLE
Add doc on OIDC-Conformant SSO

### PR DESCRIPTION
**Pull request by khorne3**

### CODE REVIEW

**Comment by khorne3** [comment]
@mpaktiti Do you mind reviewing this as well?

File: articles/sso/v2/single-page-apps-sso.md [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+This article assumes that you have already created an Auth0 Client (with its type set to **Single Page Web Applications**) with the **OIDC Conformant** flag enabled.
+:::
+
+To bypass displaying the Lock screen when logging in a user (a process known as [*silent authentication*](#silent-authentication)), you must:
+
+* Enable the **Use Auth0 instead of the IdP to do Single Sign On** flag in the [Auth0 Client's settings tab](${manage_url});
+* Have a SSO cookie for the tenant's domain (in other words, the user has previously signed in and their saved cookie is still valid);
+* Pass the name of the user's Connection to Auth0 for authentication. You can do this by:
+  * Including it as a parameter when calling the `signin` function of the [auth0.js library](/libraries/auth0js);
+  * Passing the `connection` query string parameter when calling the [Authentication API's `/authorize` endpoint](/api/authentication#login).
```
Comment (mpaktiti) [inline articles/sso/v2/single-page-apps-sso.md]: Update API URL to `/api/authentication##implicit-grant`. It's the same endpoint, but the one I'm referring to is specific to Implicit Grant.

File: articles/sso/v2/single-page-apps-sso.md [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+
+# Client-Side SSO on Single Page Applications
+
+:::panel-warning Prerequisites
+This article assumes that you have already created an Auth0 Client (with its type set to **Single Page Web Applications**) with the **OIDC Conformant** flag enabled.
+:::
+
+To bypass displaying the Lock screen when logging in a user (a process known as [*silent authentication*](#silent-authentication)), you must:
+
+* Enable the **Use Auth0 instead of the IdP to do Single Sign On** flag in the [Auth0 Client's settings tab](${manage_url});
```
Comment (mpaktiti) [inline articles/sso/v2/single-page-apps-sso.md]: I think it's better to replace the URL with sth like `${manage_url}/#/clients/${client.accountId}/settings`.

**Review by mpaktiti — CHANGES_REQUESTED** [review]
Two comments here:

1. You do not refer to the sample https://github.com/rolodato/oidc-sso-sample I think it should be added (intro text, link to download, how to run, move sample to `auth0-samples`.

2. I think you should change the doc's structure a bit, the intro to be specific. As a first thing in the doc, I would expect to see some theory (similar to What SSO Looks Like paragraph). Then a quick reference to the implementation method (silent authentication), then the prerequisites, then how to implement this, and finally a couple of screenshots (if applicable, I haven't run the sample).

File: articles/sso/v2/single-page-apps-sso.md [hunk 3]
```diff
@@ -0,0 +1,177 @@
...
+
+Suppose you have three applications:
+
+* App 1: app1.com (single page app)
+* App 2: app2.com (single page app)
+* App 3: app3.com (regular web app)
+
+If you've implemented SSO and the user logs in to *any* of the three applications, the user should automatically be logged in to the other applications.
+
+This article shows you how to implement OIDC-compliant Single Sign On via [Silent Authentication](/api-auth/tutorials/silent-authentication) for your Single Page Applications using [this sample](https://github.com/auth0-samples/oidc-sso-sample). This process involves
```
Comment (mpaktiti) [inline articles/sso/v2/single-page-apps-sso.md]: Looks like something is missing here "This process involves"

File: articles/sso/v2/single-page-apps-sso.md [hunk 4]
```diff
@@ -0,0 +1,177 @@
...
+
+### Get Started with the OIDC-Compliant Single Sign On Sample
+
+Please feel free to [download the sample](https://github.com/auth0-samples/oidc-sso-sample) and work through the examples on your local environment as you read this doc. The following instructions will help you set up the sample SPA on your local environment.
+
+<div class="alert alert-info">
+  <strong>Heads up!</strong> This document assumes that you're using port 3000 when running the sample. If you are using a different port, you'll need to adjust for this as you work through the sample (specifically the <i>auth0-variables.js</i>, <i>callback.html</i>, and <i>index.js</i> files) and configure your Auth0 Client.
+</div>
+
+1. Add `http://localhost:3000` and `http://localhost:3000/callback.html` to the Allowed Callback URLs field of your [Auth0 Client Settings](${manage_url}/#/clients/${account.clientId}/settings).
```
Comment (mpaktiti) [inline articles/sso/v2/single-page-apps-sso.md]: Missing first bullet: Create new Client (and select SPA). You can add it in the same bullet as well, I don't mind, just don't jump directly to Allowed Callback URLs.

I know you have a panel earlier, but I would remove the panel and add one extra bullet: create a client, select SPA, set the OIDC Conformant toggle to true.

File: articles/sso/v2/single-page-apps-sso.md [hunk 5]
```diff
@@ -0,0 +1,177 @@
...
+
+:::panel-warning Prerequisites
+This article assumes that you have already created an Auth0 Client (of type **Single Page Web Applications**) with the **OIDC Conformant** flag enabled.
+:::
+
+### Get Started with the OIDC-Compliant Single Sign On Sample
+
+Please feel free to [download the sample](https://github.com/auth0-samples/oidc-sso-sample) and work through the examples on your local environment as you read this doc. The following instructions will help you set up the sample SPA on your local environment.
+
+<div class="alert alert-info">
```
Comment (mpaktiti) [inline articles/sso/v2/single-page-apps-sso.md]: I think this would look better after the bullets.

File: articles/sso/v2/single-page-apps-sso.md [hunk 6]
```diff
@@ -0,0 +1,177 @@
...
+
+* App 1: app1.com (single page app)
+* App 2: app2.com (single page app)
+* App 3: app3.com (regular web app)
+
+If you've implemented SSO and the user logs in to *any* of the three applications, the user should automatically be logged in to the other applications.
+
+This article shows you how to implement OIDC-compliant Single Sign On via [Silent Authentication](/api-auth/tutorials/silent-authentication) for your Single Page Applications using [this sample](https://github.com/auth0-samples/oidc-sso-sample). This process involves
+
+## Implement SSO on Your SPA
```
Comment (mpaktiti) [inline articles/sso/v2/single-page-apps-sso.md]: I don't really see why this header is here. It has only one pabel and then it goes into another header `### Get Started with the OIDC-Compliant Single Sign On Sample`. I would add only one header with sth like `Configure the Sample Application`

File: articles/sso/v2/single-page-apps-sso.md [hunk 7]
```diff
@@ -0,0 +1,177 @@
...
+
+* Enable the **Use Auth0 instead of the IdP to do Single Sign On** flag in the [Auth0 Client's settings page](${manage_url}/#/clients/${account.clientId}/settings);
+* Have a SSO cookie for the tenant's domain (in other words, the user has previously signed in and their saved cookie is still valid);
+* Pass the name of the user's Connection to Auth0 for authentication. You can do this by:
+  * Including it as a parameter when calling the `signin` function of the [auth0.js library](/libraries/auth0js);
+  * Passing the `connection` query string parameter when calling the [Authentication API's `/authorize` endpoint](/api/authentication#implicit-grant).
+
+### Initiate a Silent Authentication Request
+
+To initiate a silent authentication request, include the `prompt` parameter in your [authorization URL](/api/authentication#implicit-grant) and set it to `none` when redirecting users to the [Authentication API's `/authorize` endpoint](/api/authentication#login).
```
Comment (mpaktiti) [inline articles/sso/v2/single-page-apps-sso.md]: Both URLs should be like the first one: /api/authentication#implicit-grant

File: articles/sso/v2/single-page-apps-sso.md [hunk 8]
```diff
@@ -0,0 +1,177 @@
...
+    client_id=4rO4qfTUOr1QafUxUGtGRYPqZfPVpuFa&
+    redirect_uri=https://YOUR_APP/callback&
+    nonce=YOUR_CRYPTOGRAPHIC_NONCE&
+    state=YOUR_OPAQUE_VALUE&
+    connection=CONNECTION_NAME&
+    prompt=none
+```
+
+:::panel-info Authorization Call Parameters
+Refer to the [tutorial on using Implict Grants](/api-auth/tutorials/implicit-grant#1-get-the-user-s-authorization) for information on the authorization call's parameters.
```
Comment (mpaktiti) [inline articles/sso/v2/single-page-apps-sso.md]: Grants -> Grant

File: articles/sso/v2/single-page-apps-sso.md [hunk 9]
```diff
@@ -0,0 +1,177 @@
...
+:::
+
+For requests received with the parameter `prompt=none`, Auth0 redirects to the `redirect_uri` specified. There are two possible outcomes:
+
+* If the user is already logged in via SSO, Auth0 sends a successful authentication response;
+* If the user is not logged in via SSO (and therefore Auth0 cannot silently authenticate the user), Auth0 sends an error response.
+
+Regardless of which outcome occurs, the sample app's `postMessage` function sends Auth0's response from the iframe back to the main page, allowing it to act based on the response.
```
Comment (mpaktiti) [inline articles/sso/v2/single-page-apps-sso.md]: Maybe add the script inline? Your call.

File: articles/sso/v2/single-page-apps-sso.md [hunk 10]
```diff
@@ -0,0 +1,177 @@
...
+
+The following is a list of possible `ERROR_CODE` values, as defined by the [OpenID Connect specification](https://openid.net/specs/openid-connect-core-1_0.html#AuthError):
+
+* `login_required`: the user isn't logged in to Auth0, so Auth0 was unable to authenticate the user;
+* `consent_required`: the user is logged in to Auth0, but hasn't consented to authorizing the client app;
+* `interaction_required`: the user is logged in to Auth0, but needs to be redirected elsewhere before Auth0 can complete authentication (for example, if a [redirect rule](/rules/redirect) exists).
+
+If your user triggers any of these errors, your app must redirect the user to the Auth0 login page so that they can authenticate using a URL without the `prompt` parameter.
+
+## Refresh Expired Access Tokens
```
Comment (mpaktiti) [inline articles/sso/v2/single-page-apps-sso.md]: I think there is too much duplication of content. Maybe remove this and add somewhere a note/panel with a link to the main silent authentication doc for details?

File: articles/sso/v2/single-page-apps-sso.md [hunk 11]
```diff
@@ -0,0 +1,177 @@
...
+---
+
+# Client-Side SSO on Single Page Applications
+
+Single Page Applications (SPA) are user-friendly apps that load a single HTML page. This page then dynamically updates as the users interacts with the app. If your SPA is associated with other apps or sites that asks for authentication, you can implement OIDC-compliant Single Sign On to minimize the number of times the user has to provide their credentials. This ensures both the security of the process and ease of use from the perspective of the user.
+
+## What SSO Looks Like
+
+Suppose you have three applications:
+
```
Comment (mpaktiti) [inline articles/sso/v2/single-page-apps-sso.md]: This is the "legacy" scenario. I think in this p2 case the sample app has only 1 SPA, right?

**Review by mpaktiti — CHANGES_REQUESTED** [review]
Hey Katie, left some comments here. You can also rename v2 to oidc and then add the document versioning functionality. You can see an example here https://github.com/auth0/docs/pull/3758/files In the p2 docs case we will name the folders legacy (the existing doc), and oidc (the p2 version of the doc, like this one).

**Pull request closed without merging**